### PR TITLE
change NIOTSNetworkEvent initializers visibility to public level

### PR DESCRIPTION
**Pull request by fananek**
_[One line description of your change]_
Change NIOTSNetworkEvent initializers visibility to public level.

### Motivation:

_[Explain here the context, and why you're making that change. What is the problem you're trying to solve.]_
Solves issue #147. Public initializers allow to extend unit test of async-http-client. For more details, see [async-http-client \#589](https://github.com/swift-server/async-http-client/issues/589) 

### Modifications:

_[Describe the modifications you've done.]_
Add public initializers for all NIOTSNetworkEvents.

### Result:

_[After your change, what will change.]_
NIOTSNetworkEvent initializers will be available across different packages.

### CODE REVIEW

**Review by fabianfett — APPROVED** [review]
Great! Thanks very much @fananek! Let's wait for @Lukasa's opinion though before we merge.

**Comment by fabianfett** [comment]
@swift-server-bot test this please

**Review by Lukasa — APPROVED** [review]
Excellent, LGTM.

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Pull request merged**
